### PR TITLE
Submit form: Support commas in tag input, mark required fields

### PR DESCRIPTION
**Pull request by lmcnulty**
Implements all of the changes named in #1112.

- In tag inputs, a comma now begins a new tag, and a pasted comma-separated list get split into its components.
- Adds an asterisk before required fields.
- Hides the entity fields when .
- Sets the max description length to 500.

### CODE REVIEW

File: site/gatsby-site/src/components/forms/TextInputGroup.js [hunk 1]
```diff
@@ -14,31 +14,40 @@ const TextInputGroup = ({
...
-      </InputGroup>
-    </Form.Group>
-  </div>
-);
+}) => {
+  const [optional, setOptional] = useState(true);
+
+  useEffect(async () => {
+    setOptional(await schema.fields[name].isValid(undefined));
+  }, []);
```
Comment (cesarvarela) [inline site/gatsby-site/src/components/forms/TextInputGroup.js:25]: Not sure I follow this 🤔

**Review by cesarvarela — COMMENTED** [review]
One question before merging

File: site/gatsby-site/src/components/forms/TextInputGroup.js (hunk 1 above)
Comment (lmcnulty) [reply site/gatsby-site/src/components/forms/TextInputGroup.js:25]: `schema` contains the full `yup` schema. We descend into the field matching the current input group, then we test if the field is optional according to the schema by seeing if `isValid` returns `true` given `undefined`. For some reason, `isValid` is asynchronous, so we need to wrap it in a `useEffect`.

Comment (cesarvarela) [reply site/gatsby-site/src/components/forms/TextInputGroup.js:25]: Oh my! There must be a better way to do this, but let's go with it for now. Forms need a rework anyway, including moving them to flow bite/tailwind.

**Comment by cesarvarela** [comment]
@lmcnulty since you are at it:

`author` and `date_downloaded` show an asterisk but no invalid feedback message:

image

*This is by pressing `Submit`)

**Comment by lmcnulty** [comment]
@cesarvarela I fixed the validation for tag inputs. The "Date Downloaded" field doesn't show an error because it defaults to the current day as of the last PR.



While I was there I also fixed the corner weirdness from #943.
<table>
  <tr><th>Before</th><th>After</th>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>

**Comment by cesarvarela** [comment]
@lmcnulty please merge staging, tests should pass now

**Pull request merged**